### PR TITLE
Introducing clang-tidy config options file on the command line interface.

### PR DESCRIPTION
**Pull request by bruntib**

### CODE REVIEW

File: libcodechecker/analyze/analyzers/analyzer_types.py [hunk 1]
```diff
@@ -384,6 +384,20 @@ def __build_clang_tidy_config_handler(args, context):
         # No clang tidy arguments file was given in the command line.
         LOG.debug_analyzer(aerr)
 
+    try:
+        # The config file dumped by clang-tidy contains "..." at the end. This
+        # has to be emitted, otherwise -config flag of clang-tidy can't consume
+        # it.
+        with open(args.tidy_config, 'rb') as tidy_config:
```
Comment (gyorb) [inline libcodechecker/analyze/analyzers/analyzer_types.py]: please use `io.open` to open files

File: libcodechecker/libhandlers/analyzers.py [hunk 2]
```diff
@@ -86,6 +106,17 @@ def main(args):
             analyzer_types.supported_analyzers,
             context)
 
+    if args.dump_config:
+        binary = context.analyzer_binaries.get(args.dump_config)
+
+        if args.dump_config == 'clang-tidy':
+            subprocess.call([binary, '-dump-config', '-checks=*'])
+        elif args.dump_config == 'clangsa':
+            # TODO: Not supported by ClangSA yet!
```
[the remark this answers is not in the document]
Comment (gyorb) [reply libcodechecker/libhandlers/analyzers.py]: Shouldn't we log a warning message here to the user, that this is not supported yet?

File: libcodechecker/analyze/analyzers/analyzer_types.py [hunk 3]
```diff
@@ -384,6 +384,20 @@ def __build_clang_tidy_config_handler(args, context):
...
+            lines = tidy_config.readlines()
+            lines = filter(lambda x: x != '...\n', lines)
+            config_handler.checker_config = ''.join(lines)
```
Comment (gyorb) [inline libcodechecker/analyze/analyzers/analyzer_types.py]: maybe `''.join(lines[:-1])` would be enough too if we know that only the last line is not needed?

Comment (bruntib) [reply libcodechecker/analyze/analyzers/analyzer_types.py]: Me too, I was thinking of this. Currently the dump emitted by clang-tidy contains these three dots and an empty line at the end. So actually `lines[:-2]` should be used. This is a somewhat more general solution, if we don't want to rely on the existence of the last empty line. Let's discuss once more which solution should be used.

File: libcodechecker/libhandlers/analyzers.py [hunk 4]
```diff
@@ -86,6 +106,17 @@ def main(args):
...
+            LOG.warning("'--dump-config clangsa' is not supported yet.")
```
Comment (csordasmarton) [inline libcodechecker/libhandlers/analyzers.py]: For me this message was never be printed.
I checked the [logger.conf](https://github.com/Ericsson/codechecker/blob/master/config/logger.conf) file and it doesn't contain a `logger` for `analyze` type:
https://github.com/Ericsson/codechecker/blob/0df57dcf8fa83bc5cb4fac5a779bd109d6b2d32d/libcodechecker/libhandlers/analyzers.py#L22

File: docs/user_guide.md [hunk 5]
```diff
@@ -1133,6 +1141,16 @@ optional arguments:
...
+  --dump-config {clangsa,clang-tidy}
+                        Dump the available checker options for the given
+                        analyzer to the standard output. Currently only clang-
+                        tidy supports this option. The output can be
+                        redirected to a file named .clang-tidy. If this file
+                        is placed to the project directory then the options
+                        are applied to the files under that directory. This
+                        config file can also be provided via 'CodeChecker
+                        analyze' and 'CodeChecker check' commands. (default:
+                        None)
```
Comment (gyorb) [inline docs/user_guide.md]: the `(default: None)` can be removed it is not printed.

**Pull request merged**